addons/stock/models/stock_move.py

```diff
@@ -1014,10 +1014,12 @@ def _keys_out_sorted(ml):
     def _action_cancel(self):
         if any(move.state == 'done' for move in self):
             raise UserError(_('You cannot cancel a stock move that has been set to \'Done\'.'))
+        self = self.filtered(lambda x: x.state != 'cancel')
```

Comment:
But in last line of method you have `self.write({'state': 'cancel', 'move_orig_ids': [(5, 0, 0)]})`, outside the loop. So before this patch canceled moves passed through this `write` and with this patch they will not pass through it. You won't assure that canceled moves don't have any move_orig_ids.

Reply:
Indded

Reply:
I think the proper way to pass the move state in cancel is to pass inside the _action_cancel function so it should not be a problem.

Reply:
I changed it anyway, this way we don't change behavior in stable